upgrade/lib/validation/CASMREL-776-CSM12-NCN-boot-order-backport/scripts/metal-lib.sh

```diff
@@ -0,0 +1,441 @@
+                    ;;
+                ncn-w*)
+                    efibootmgr | grep -i 'port 1' | grep -i 'pxe ipv4' | tee /tmp/bbs1
+                    efibootmgr | grep cray | tee /tmp/bbs2
+                    ;;
+                *)
+                    efi_fail_host
+                    ;;
+            esac
+            ;;
+        *'Intel'*'Corporation'*)
+            # Removal file(s) ...
+            efibootmgr | grep -vi 'ipv4' | grep -iP '(sata|nvme|uefi)' | tee /tmp/rbbs1
+            efibootmgr | grep -i baseboard | tee /tmp/rbbs2
+            efibootmgr_prefix=''
+            efi_trim
+            efi_specials
+            efi_remove
+            case $hostname in
+                ncn-m*)
+                    efibootmgr | grep -i 'ipv4' | grep -iv 'baseboard' | tee /tmp/bbs1
+                    efibootmgr | grep -i 'cray' | tee /tmp/bbs2
+                    ;;
+                ncn-s*)
+                    efibootmgr | grep -i 'ipv4' | grep -iv 'baseboard' | tee /tmp/bbs1
+                    efibootmgr | grep -i 'cray' | tee /tmp/bbs2
+                    ;;
+                ncn-w*)
+                    efibootmgr | grep -i 'ipv4' | grep -iv 'baseboard' | tee /tmp/bbs1
+                    efibootmgr | grep -i 'cray' | tee /tmp/bbs2
+                    ;;
+                *)
+                    echo >&2 $0 Unsupported node name $hostname
+                    return 1
+                    ;;
+            esac
+            ;;
+        *)
+            echo >&2 not modifying unknown vendor: $vendor
+            return 1
+            ;;
+    esac
+
+    [ "$no_enforce" = 0 ] && efi_enforce
+
+    mprint "log file located at $0.log"
+}
+
+function paginate() {
+    local url="$1"
+    local token
+    { token="$(curl -sSk "$url" | tee /dev/fd/3 | jq -r '.continuationToken // null')"; } 3>&1
+    until [[ "$token" == "null" ]]; do
+        {
+            token="$(curl -sSk "$url&continuationToken=${token}" | tee /dev/fd/3 | jq -r '.continuationToken // null')";
+        } 3>&1
+    done
+}
+
+function install_csm_rpms() {
+
+    # Verify nexus is available.  It's expected to *not* be available during initial install of the NCNs.
+    if ! curl -sSf https://packages.local/service/rest/v1/components?repository=csm-sle-15sp3 >& /dev/null; then
+        echo "unable to contact nexus, bailing"
+        return 0
+    fi
+
+    # Retreive the packages from nexus
+    goss_servers_url=$(paginate "https://packages.local/service/rest/v1/components?repository=csm-sle-15sp2" \
+        | jq -r  '.items[] | .assets[] | .downloadUrl' | grep goss-servers | sort -V | tail -1)
+    csm_testing_url=$(paginate "https://packages.local/service/rest/v1/components?repository=csm-sle-15sp2" \
+        | jq -r  '.items[] | .assets[] | .downloadUrl' | grep csm-testing | sort -V | tail -1)
+    zypper install -y $goss_servers_url $csm_testing_url && systemctl enable goss-servers && systemctl restart goss-servers
+}
```

Comment:
you're technically missing the update from brad here - to include platform-utils - but I don't think that matters for this hotfix usecase.